core/src/test/java/org/elasticsearch/gateway/PrimaryShardAllocatorTests.java

```diff
@@ -193,6 +198,74 @@ public void testFoundAllocationAndAllocating() {
     }
 
     /**
+     * Tests that when the nodes with prior copies of the given shard all return a decision of NO, but
+     * {@link AllocationDecider#canForceAllocatePrimary(ShardRouting, RoutingNode, RoutingAllocation)}
+     * returns a YES decision for at least one of those NO nodes, then we force allocate to one of them
+     */
+    public void testForceAllocatePrimary() {
+        testAllocator.addData(node1, ShardStateMetaData.NO_VERSION, "allocId1", randomBoolean());
+        AllocationDeciders deciders = new AllocationDeciders(Settings.EMPTY, new AllocationDecider[] {
+            // since both deciders here return a NO decision for allocating a shard,
+            // the allocator will see if it can force assign the primary, where the decision will be YES
+            new TestAllocateDecision(Decision.NO), getNoDeciderThatAllowsForceAllocate()
```

Comment:
does it make sense to replace `new TestAllocateDecision(Decision.NO)` by `new TestAllocateDecision(randomBoolean() ? Decision.YES : Decision.NO)`.